cardano-api/src/Cardano/Api/LocalStateQuery.hs

```diff
@@ -149,6 +150,24 @@ queryStakeDistributionFromLocalState network socketPath point = do
       socketPath
       pointAndQuery
 
+queryLocalLedgerState
+  :: blk ~ ShelleyBlock TPraosStandardCrypto
+  => Network
+  -> SocketPath
+  -> Point blk
+  -> ExceptT LocalStateQueryError IO LByteString
```

Comment:
Should the return type here be `LByteString` or something else?

Reply:
Why can't we get a `LedgerState` here?

Reply:
Okay, I just read the documentation here:

https://github.com/input-output-hk/ouroboros-network/blob/9d9754c9ddcfff82b27c371a545aa4680d86d996/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs#L320-L333

Should we attempt the `decodeFull` here and return the `LedgerState` on success?

Reply:
We need to be able to return the decoded `Ledger.LedgerState TPraosStandardCrypto` if we can and if that fails return the `LByteString`. I could do that with an `Either` but :shrug: .

Reply:
Changed the type of that so its:
```
ExceptT LocalStateQueryError IO (Either LByteString (Ledger.LedgerState TPraosStandardCrypto))
```
which I think is a minor improvement.

Reply:
> Should we attempt the decodeFull here and return the LedgerState on success?

If decide fails we want to do a generic CBOR decode.